### PR TITLE
Tests, enhancements and fixes

### LINKED ISSUE #23: Check the functionality of each system now and check if it's working as expected. (main.py, security_system.py, train_model.py and tests)
### Description

Conduct a comprehensive review and testing of the system components, including main.py, security_system.py, train_model.py, and the associated test cases. The primary objective is to ensure that each component functions as expected, identify and report any bugs, and suggest possible improvements while performing a thorough code refactoring.

### Expected Behavior

The components are working without any issues and generate expected results.

### Additional Context

Refer to the following comment: https://github.com/ContriHUB/SecureFace/pull/22#issuecomment-1762637226

### PR DESCRIPTION
**Pull request by Aqua-16**
## Description

Updated all files with some minor changes, refactors and fixes, mostly for accommodating Eigen and Fisher predictions.
Also updated the Project structure to reflect current state.

## Related Issue

Fixes #23 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

I have tested it using unittest as well as random images I acquired. I have also tested the live webcam and it works perfectly fine.

## Screenshots (if appropriate):

image
image
image
image

## Checklist:

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: train_model/train_model.py
Comment (sanskaromar) [inline train_model/train_model.py:1]: We can also include which algorithm was used in the training summary.

Comment (sanskaromar) [reply train_model/train_model.py:1]: Remove all comments with #TODO and fix spacing there as well. Remove unnecessary comments/rewrite comments.

Comment (Aqua-16) [reply train_model/train_model.py:1]: That's a good idea!

Comment (Aqua-16) [reply train_model/train_model.py:1]: Fixed!

File: app/security_system.py [hunk 1]
```diff
@@ -8,40 +8,43 @@ class SecuritySystem:
...
         else:
             self.cascade_path = cascade_path
+            
         # Load authorized persons from CSV
+        
         self.authorized_persons = self.load_authorized_persons()
+        
         # Load label-to-name mapping from CSV
+        
         self.label_to_name = self.load_label_to_name(algorithm)
```
Comment (sanskaromar) [inline app/security_system.py]: No need for spacing after each comment.

Comment (Aqua-16) [reply app/security_system.py]: I did that because I thought it looked good. Should I change it?

Comment (Aqua-16) [reply app/security_system.py]: Had to fix a case issue and tense problem, so fixed the spaces too. Left space on top margin and removed the bottom one to follow the same look as other comments.

**Review by sanskaromar — APPROVED** [review]
🚀

**Pull request merged**